lib/place.ml

```diff
@@ -172,51 +176,53 @@ let print_html_places_surnames conf base (array : (string list * (string * iper
   loop [] (Array.to_list array) ;
   Wserver.printf "</ul>\n"
 
-let print_all_places_surnames_short conf base ~add_birth ~add_baptism ~add_death ~add_burial =
+let print_aux_opt ~add_birth ~add_baptism ~add_death ~add_burial ~add_marriage =
+    (if add_birth then "&bi=on" else "") ^
+    (if add_baptism then "&bp=on" else "") ^
+    (if add_death then "&de=on" else "") ^
+    (if add_burial then "&bu=on" else "") ^
+    (if add_marriage then "&ma=on" else "")
+
+let print_aux conf title fn =
+  Hutil.header conf title;
+  Hutil.print_link_to_welcome conf true;
+  fn () ;
+  Hutil.trailer conf
+
+let print_all_places_surnames_short conf base ~add_birth ~add_baptism ~add_death ~add_burial ~add_marriage =
   let array =
     get_all
-      conf base ~add_birth ~add_baptism ~add_death ~add_burial
+      conf base ~add_birth ~add_baptism ~add_death ~add_burial ~add_marriage
       "" 0
       fold_place_short
       (fun _ -> true)
       (fun prev _ -> match prev with Some n -> n + 1 | None -> 1)
       (fun x -> x)
+      max_int
   in
-  let title _ = Wserver.printf "%s" (capitale (transl conf "place")) in
   Array.sort (fun (s1, _) (s2, _) -> Gutil.alphabetic_order s2 s1) array ;
-  let add_birth = p_getenv conf.env "bi" = Some "on" in
-  let add_baptism = p_getenv conf.env "bp" = Some "on" in
-  let add_death = p_getenv conf.env "de" = Some "on" in
-  let add_burial = p_getenv conf.env "bu" = Some "on" in
-  let add_marriage = p_getenv conf.env "ma" = Some "on" in
-  let opt =
-    (if add_birth then "&bi=on" else "") ^
-    (if add_baptism then "&bp=on" else "") ^
-    (if add_death then "&de=on" else "") ^
-    (if add_burial then "&bu=on" else "") ^
-    (if add_marriage then "&ma=on" else "")
-  in
-  Hutil.header conf title;
-  Hutil.print_link_to_welcome conf true;
-  Wserver.printf
-    "<p><a href=\"%sm=PS%s&k=\">%s</a></p><p>"
-    (commd conf) opt (transl conf "long display") ;
-  Array.iter
-    (fun (s, x) ->
-       Wserver.printf
-         "<a href=\"%sm=PS%s&k=%s\">%s</a> (%d),\n"
-         (commd conf) opt (Util.code_varenv s) s x)
-    array ;
-  Wserver.printf "</p>\n";
-  Hutil.trailer conf
+  let title _ = Wserver.printf "%s" (capitale (transl conf "place")) in
+  print_aux conf title begin fun () ->
+    let opt = print_aux_opt ~add_birth ~add_baptism ~add_death ~add_burial ~add_marriage in
+    Wserver.printf
+      "<p><a href=\"%sm=PS%s&display=long\">%s</a></p><p>"
+      (commd conf) opt (transl conf "long display") ;
+    Array.iter
+      (fun (s, x) ->
+         Wserver.printf "<a href=\"%sm=PS%s&k=%s\">%s</a> (%d),\n"
```

Comment:
Could you please remove the last comma ?

Reply:
For sure.